### ISSUE TITLE
Configuring helm for more than 2 minutes

### ISSUE THREAD

**Issue by rafaelgp** [syl20bnr/spacemacs#2705]
Hello,

I started using Linux in a Virtual Box in order to get the most of Emacs.
I'm quite happy with the change but when I open Emacs I have to wait more than 2 minutes to start typing anything.

Spacemacs loads very quick. Less than 12 seconds:

![configuring](http://clients.tourvista.co.uk/vt/zip/tmp/configuring.jpg)

But then I get the message `configuring package helm...` and Emacs freezes for 2 minutes until it finishes. As you can see in the image below, it takes 130 seconds!

![boot](http://clients.tourvista.co.uk/vt/zip/tmp/boot-log.jpg)

Has anyone experienced this problem?

VirtualBox 5.0
Debian Stretch (Testing) with LXDE
Emacs 24.5.1
Spacemacs 0.103.5

Thanks

**Comment by TheBB** [comment]
Yes, I have a similar problem. Some things take about two minutes to start, and this includes

* Helm
* Emacs itself
* Firefox
* Thunderbird

The two minutes figure is relatively exact and reproducible.

It only happens on my home computer, despite my work computer having a very similar system (same OS). Firefox and Thunderbird show two started processes during the two minutes, one of them defunct. Emacs does not.

In short I have no idea what causes this except it's probably not an Emacs or Spacemacs problem, at least not in my case. It's been very frustrating and I've been debating whether to reformat and reinstall the OS. Only thing stopping me is probably that I only have to start those programs a few times a week.

**Comment by justbur** [comment]
It's probably a network thing like tramp timing out or a bad file path somewhere like your history. I'd like to know more about why helm is way slower to load on some systems but haven't been able to figure it out.

**Comment by justbur** [comment]
@rafaelgp Here's a simple test to try and maybe we can narrow it down. Put this at the top of your .emacs.d/init.el

```elisp
(defadvice require (around timed-require activate)
  (let ((start (current-time))
        res)
    (setq res ad-do-it)
    (message "Required %s: %s sec" (ad-get-arg 0) (float-time (time-since start)))
    res))
```

It's going to spam your messages buffer with Required package: x sec, but it might allow you to narrow down the problem to an offending file. This is not foolproof, because there are other ways files can be loaded (through "load" for instance).

**Comment by justbur** [comment]
Actually here's a more useful version

```elisp
(defadvice require (around my-require activate)
  (let ((start (current-time))
        res delta)
    (setq res ad-do-it)
    (setq delta (float-time (time-since start)))
    (when (> delta 0.001)
      (message "Required %s: %s" (ad-get-arg 0) delta))
    res))
```

**Comment by rafaelgp** [comment]
Thank you very much for you quick reply. Very much appreciate it.
Here is the output of your second code:

![Waiting for tram](http://clients.tourvista.co.uk/vt/zip/tmp/wait-tramp.jpg)

Looks like `tramp-loaddefs` is the one delaying everything

**Comment by TheBB** [comment]
Maybe this discussion has something for you https://github.com/emacs-helm/helm/issues/1000

**Comment by rafaelgp** [comment]
Boom!

After adding the line below, Spacemas loads in 7.5 seconds and helm in 1.5 seconds. 

```
(setq tramp-ssh-controlmaster-options "-o ControlMaster=auto -o ControlPath='tramp.%%C' -o ControlPersist=no")
```

Now we know that both Arch and Debian testing users are affected by this.

**Comment by justbur** [comment]
@syl20bnr maybe we could add something to deal with this in .spacemacs? If you don't use tramp you can also disable it completely with 

(setq tramp-mode nil)

More [here](https://www.gnu.org/software/emacs/manual/html_node/tramp/Frequently-Asked-Questions.html)
